autobahn/wamp_subscribe_options.ipp

```diff
@@ -83,12 +91,12 @@ struct pack<autobahn::wamp_subscribe_options>
             msgpack::packer<Stream>& packer,
             autobahn::wamp_subscribe_options const& options) const
     {
-        std::map<std::string, std::string> options_map;
+        std::map<std::string, msgpack::object> options_map;
         bool should_pack_options = false;
 
         if (options.is_match_set())
         {
-            options_map["match"] = options.match();
+            options_map["match"] = msgpack::object(options.match());
```

Comment:
I have no clue how and why msgpack accepts a string without zone, but this seems unsafe and might lead to memory corruption because the string memory is not kept around by anyone. Ditto the change below.